source/jormungandr/tests/chaos_disruptions_tests.py

```diff
@@ -224,6 +219,121 @@ def test_disruption_on_journey(self):
 
             assert any(d['uri'] == 'bob_the_disruption' for d in disruptions)
 
+@dataset([("main_routing_test", ['--BROKER.rt_topics='+chaos_rt_topic, 'spawn_maintenance_worker'])])
+class TestChaosDisruptionsBlocking(ChaosDisruptionsFixture):
+    """
+    Note: it is done as a new fixture, to spawn a new kraken, in order not the get previous disruptions
+    """
+
+    def __init__(self):
+        ChaosDisruptionsFixture.__init__(self)
+        self.nb_disruptions_map = {}
+
+    def get_nb_disruptions(self):
+        def set_nb_disruptions(l):
+            for i in l:
+                if not "disruptions" in i:
+                    self.nb_disruptions_map[i['id']] = 0
+                    continue
+                self.nb_disruptions_map[i['id']] = len(i['disruptions'])
+
+        response = self.query_region("routes")
+        set_nb_disruptions(response['routes'])
+        response = self.query_region("lines")
+        set_nb_disruptions(response['lines'])
+        response = self.query_region('networks')
+        set_nb_disruptions(response['networks'])
+
+    def run_t_(self, object_id, type_):
+        disruption_uri = "blocking_{}_disruption".format(type_)
+        nb_disruptions = self.nb_disruptions_map[object_id]
+        # We send a blocking disruption on line A
+        self.send_chaos_disruption_and_sleep(disruption_uri, object_id, type_, blocking=True)
+        self.get_nb_disruptions()
```

Comment:
I'm not a fan of the side effect there, wouldn't it be clearer to return the map there and not store it in self ?